File: src/types/budget.ts
export type Budget = {
  id: string;
  userId: string;
  categoryId?: string;
  categoryName?: string;
  amount: number;
  spentAmount: number;
  remainingAmount: number;
  usagePercentage: number;
  month: number;
  year: number;
  createdAt?: string;
  updatedAt?: string;
};

export type BudgetInput = {
  amount: number;
  categoryId?: string;
  month: number;
  year: number;
};

export type BudgetListData = {
  items: Budget[];
  totalCount: number;
};

File: Backend/FinancialAdvisor.Api/Models/Budget.cs
namespace FinancialAdvisor.Api.Models;

public sealed class Budget
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public Guid? CategoryId { get; set; }
    public decimal Amount { get; set; }
    public int Month { get; set; }
    public int Year { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public User User { get; set; } = null!;
    public Category? Category { get; set; }
}
